### PR TITLE
ExeUnit: VM flow

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Changes:
* fix: `Run` arguments are converted into `argv`
* fix: add `golem.runtime.name` and `golem.runtime.version` to `Demand` objects in `requestor-sdk`
* fix: computation termination condition in `requestor-sdk`
* refactor: use a common `DataDir` implementation in `serv` and `ya-provider`

Requires https://github.com/golemfactory/ya-runtime-vm/pull/48

### CODE REVIEW

File: agent/requestor-sdk/src/requestor.rs [hunk 1]
```diff
@@ -349,22 +370,37 @@ impl Requestor {
                 break;
             }
 
-            let mut res = activity_api
+            all_res = activity_api
```
Comment (etam) [inline agent/requestor-sdk/src/requestor.rs:373]: I think the `let` keyword should stay here and the `all_res` defined outside of the loop should be deleted. Unless there's anything in it that must survive from one iteration to next.

File: agent/requestor-sdk/Cargo.toml [hunk 2]
```diff
@@ -22,3 +22,6 @@ url = "2.1.1"
 ya-agreement-utils = "0.1"
 ya-client = "0.3"
 sha3 = "0.9.1"
+
+[dev-dependencies]
+structopt = "0.3.15"
```
[the remark this answers is not in the document]
Comment (mfranciszkiewicz) [reply agent/requestor-sdk/Cargo.toml:27]: There is no guideline. It's a common version found in Cargo.toml files in this project. I don't believe that this PR should introduce any kind of guidelines or unification

File: agent/requestor-sdk/examples/run_vm_task.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        match args {
+            Location::Local { path } => Package::Archive(path),
+            Location::Url { digest, url } => Package::Url { digest, url },
+        }
+    }
+}
+
+#[actix_rt::main]
+async fn main() -> Result<()> {
+    let _ = dotenv::dotenv()?;
```
[the remark this answers is not in the document]
Comment (mfranciszkiewicz) [reply agent/requestor-sdk/examples/run_vm_task.rs:34]: The skeleton of this file has been copied from the `run_task` example.

I believe that the author's intention was to highlight that `dotenv::dotenv()?` returns a value other than `()` and which we don't care about.

File: agent/requestor-sdk/src/requestor.rs (hunk 1 above)
Comment (mfranciszkiewicz) [reply agent/requestor-sdk/src/requestor.rs:373]: See line 394

**Review by etam — APPROVED** [review]
ok

**Review by etam — APPROVED** [review]

**Pull request merged**